updating the compatibility docs for aspnet & aspnetcore to include CL… (#9600)
DataDog/documentation: content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md

## Compatibility

- The .NET Tracer supports all .NET-based languages (for example, C#, F#, Visual Basic).

- The .NET Tracer supports automatic instrumentation on .NET 5, .NET Core 3.1, and .NET Core 2.1. It also supports [.NET Framework][1].

- The .NET Tracer library for Datadog is open-source. For more information see the [tracer Github repository][2].

<div class="alert alert-warning"> 

>   <strong>Notes:</strong><br><br>- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (for example, APM). Datadog strongly recommends only running a single APM solution within your applications' environment to ensure maximum visibility.<br><br>- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.

Reviewer: ```suggestion
  <strong>Notes:</strong><br><ul><li>Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution within your application environment.</li><li> If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.</li></ul>
```

The .NET Tracer supports automatic instrumentation on .NET 5, .NET Core 3.1, and .NET Core 2.1. It also supports [.NET Framework][2].

<div class="alert alert-warning"> 

>   <strong>Notes:</strong><br>- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (for example, APM). Datadog strongly recommends only running a single APM solution within your applications' environment to ensure maximum visibility.<br>- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.

Reviewer 2: Formatting nit: shouldn't this be a real unsorted list (that renders to `<ul><li>`) instead of faking it with `<br>-`?
```suggestion
  <strong>Notes:</strong>
- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (for example, APM). Datadog strongly recommends only running a single APM solution within your applications' environment to ensure maximum visibility.
- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.
```

Reviewer 2: Please ignore if you tried this already and it doesn't work inside a `<div>` 😅

Author: Moving our conversation from Slack to the PR. 

This will break the font layout of the alert box.

>   <strong>Notes:</strong><br>- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (e.g. APM product). We strongly recommend only running a single APM solution within your applications' environment to eliminate the risk or laps of visibility.<br>- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.

Reviewer 3: ```suggestion
  <strong>Notes:</strong><br>- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (for example, APM). Datadog strongly recommends only running a single APM solution within your applications' environment to eliminate the risk of laps in visibility.<br>- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.
```

Reviewer 4: A slight edit to what @cswatt is suggesting, I propose:
```suggestion
  <strong>Notes:</strong><br>- Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API only allows one subscriber (for example, APM). Datadog strongly recommends only running a single APM solution within your applications' environment to ensure maximum visibility.<br>- If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.
```
